cpp/src/arrow/dataset/file_base.cc

```diff
@@ -115,7 +110,7 @@ Result<std::shared_ptr<FileFragment>> FileFormat::MakeFragment(
 }
 
 // TODO(ARROW-12355[CSV], ARROW-11772[IPC], ARROW-11843[Parquet]) The following
```

Comment:
I updated the comment.  Making it pure virtual failed because of some test utilities (DummyFileFormat / JSONRecordBatchFormat) but, more significantly, I think this is being used by the PR to add ORC support to datasets (#10991).

Reply:
Ah, fair point. Thanks for looking.